src/routes/pdf-fetcher/containers/PdfFetcherPage.js

```diff
@@ -31,20 +31,14 @@ class PdfFetcherPage extends React.Component {
   }
 
   componentWillMount () {
-    const hierarchy = new Hierarchy(this.props.path)
-
-    const error = hierarchy.build(this.props.pages)
-    if (error) {
-      return <Error error={error}/>
-    }
-
-    this.fetchUrl(hierarchy.top())
+    this.fetchUrl(CategoryModel.getCategoryByPath(this.props.categories, this.props.path))
   }
 
-  addPageIdsRecursively (pageIds, parentPage) {
-    parentPage.children.forEach((page) => {
-      pageIds.push(page.numericId)
-      this.addPageIdsRecursively(pageIds, page)
+  addCategoryIdsRecursively (categoryIds, currentCategory) {
```

Comment:
Also a usecase for https://www.npmjs.com/package/un-flatten-tree

Reply:
we also need recursion here if we have a flattened tree